google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfiguration.java

```diff
@@ -180,15 +187,39 @@ private void checkCommonConfig(AppEngineDeployable deployable) throws RuntimeCon
    *
    * @param deployable the {@link AppEngineDeployable deployment source} that was selected by the
    *     user to deploy
+   * @param project the {@link Project} that this configuration belongs to
    * @throws RuntimeConfigurationError if this configuration is not valid for a flex deployment
    */
-  private void checkFlexConfig(AppEngineDeployable deployable) throws RuntimeConfigurationError {
+  private void checkFlexConfig(AppEngineDeployable deployable, Project project)
+      throws RuntimeConfigurationError {
     check(
         !(deployable instanceof UserSpecifiedPathDeploymentSource)
             || (!StringUtil.isEmpty(userSpecifiedArtifactPath)
                 && isJarOrWar(userSpecifiedArtifactPath)),
         "appengine.flex.config.user.specified.artifact.error");
-    check(!StringUtils.isBlank(moduleName), "appengine.flex.config.browse.app.yaml");
+    check(!StringUtils.isBlank(moduleName), "appengine.flex.config.select.module");
+
+    AppEngineFlexibleFacet facet = AppEngineFlexibleFacet.getFacetByModuleName(moduleName, project);
+    check(facet != null, "appengine.flex.config.select.module");
+
+    String appYamlPath = facet.getConfiguration().getAppYamlPath();
+    check(!StringUtils.isBlank(appYamlPath), "appengine.flex.config.browse.app.yaml");
+    check(Files.exists(Paths.get(appYamlPath)), "appengine.deployment.config.appyaml.error");
+
+    try {
+      Optional<FlexibleRuntime> runtime =
+          AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(appYamlPath);
+      if (runtime.isPresent() && runtime.get().isCustom()) {
```

Comment:
I wish but since the `check()` methods throw an exception, it's not accepted by the signatures of the `@FunctionalInterface` parameters since they don't throw exceptions.